lib/definition/types/lambda.rb

```diff
@@ -1,48 +1,65 @@
 # frozen_string_literal: true
 
 require "definition/types/base"
+require "definition/types/error_renderers/lambda"
 
 module Definition
   module Types
     class Lambda < Base
-      module Dsl
-        def conform_with(value)
-          ConformResult.new(value)
-        end
-
-        def fail_with(error_message)
-          self.error_message = error_message
-        end
-      end
-      include Dsl
+      attr_accessor :test_lambda
```

Comment:
Can you explain why it's called `test_lambda` please?
Is it because we are going to assess / test whether it's result yields any errors?
This name somehow has a notion of either
- being a flag (like test a lambda or not)
- it just contains "test" data (not in the sense of assessing it's result)

Maybe a name like `lambda_block` would also suite here?

Reply:
named it like this because its the lambda that the user defined and that is used to test values during conforming. maybe somthing like user_lambda_block or so would be less confusing?

Reply:
> maybe somthing like user_lambda_block or so would be less confusing? 

Sounds good!
Not sure about `user_~` since it's not about the typical "user" but the developer (OK .. the user of the gem).
But something in this direction would be awesome.